### PR TITLE
feat: update policies enhancements

### PR DESCRIPTION
**Pull request by Julusian**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

* **What is the current behavior?** (You can also link to an open issue here)

The update-policies/sync-ingest-changes does not allow for some cases. Such as updating infinites, or adlibs which were executed outside of their part

* **What is the new behavior (if this is a feature change)?**

This allows both those cases to be handled properly

* **Other information**:

Replaces https://github.com/nrkno/tv-automation-sofie-blueprints-integration/pull/83

**Status**

- [ ] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [x] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

File: meteor/lib/collections/PieceInstances.ts [hunk 1]
```diff
@@ -35,8 +35,19 @@ export function protectPieceInstance(pieceInstance: IBlueprintPieceInstance): De
 
 export type PieceInstancePiece = Omit<Piece, 'startRundownId' | 'startSegmentId'>
 
+export interface PieceInstanceInfinite
+	extends ProtectedStringProperties<Required<IBlueprintPieceInstance>['infinite'], 'infinitePieceId'> {
+	/** A random id for this instance of this infinite */
+	infiniteInstanceId: PieceInstanceInfiniteId
+
+	// /** The first partInstance this existed in */
+	// firstPartInsanceId: PartInstanceId
```
Comment (nytamin) [inline meteor/lib/collections/PieceInstances.ts:44]: typo and commented out, remove?

File: meteor/server/api/blueprints/context/syncIngestUpdateToPartInstance.ts [hunk 2]
```diff
@@ -39,13 +55,54 @@ export class SyncIngestUpdateToPartInstanceContext extends RundownContext
...
 		this._partInstanceCache.updateOtherCacheWithData(cache.PartInstances)
 	}
 
+	syncPieceInstance(
+		pieceInstanceId: string,
+		modifiedPiece?: Omit<IBlueprintPiece, 'lifespan'>
+	): IBlueprintPieceInstance {
+		const proposedPieceInstance = this._proposedPieceInstances.get(protectString(pieceInstanceId))
+		if (!proposedPieceInstance) {
+			throw new Error('PieceInstance could not be found')
```
Comment (nytamin) [inline meteor/server/api/blueprints/context/syncIngestUpdateToPartInstance.ts:73]: This error would benefit from adding the not-found pieceInstanceId. it should also be a Meteor.error

**Review by nytamin — APPROVED** [review]
Looks pretty good. I haven't tested it locally but the ideas seem sound.

A couple comments and minor notes:
* style note: There are a few places where code has been commented out instead of removed.
* There are a lot of CircleCI config changes, which I don't see directly relates to the issue described in the PR description. Perhaps keep such things separate next time? :)

**Comment by Julusian** [comment]
I forgot to do a pass of going through the code to remove bits I commented out. I have done that, and the circleci changes have been pulled out too

**Pull request merged**